### PR TITLE
fix/use supported urls field

### LINKED ISSUE #12152: Mistral OCR with Mastra errors
This issue was created from Discord post 1463524876253139109:

[Open in Discord](https://discord.com/channels/1309558646228779139/1463524876253139109)

Hi,
I'm encountering an issue when trying to process a PDF document through the API.
I'm generating a signed URL for my PDF file (hosted on cloud storage) and passing it via the file content block with the data field. However, the API seems to interpret it as raw bytes rather than a URL.
Here's my current approach:
```ts
// it's an agent with mistral-large-latest
const fileUrl = await drive.use().getSignedUrl(fileName, {
  expiresIn: '5m',
})
const mediaType = this.getMimeType(document)
console.log('Generated signed URL for document:', fileUrl)

const stream = await agent.stream([
  {
    role: 'user',
    content: [
      {
        type: 'text',
        text: 'Lorem ipsum...',
      },
      {
        type: 'file',
        data: fileUrl,
        mediaType,
      },
    ],
  },
], {
  structuredOutput: {
    schema: templateGeneratorOutputSchema,
  },
  modelSettings: {
    temperature: 0.3,
    maxOutputTokens: 8192,
  },
})
```

```
request_id: "revle95fo7wx871avpuuxdfv"
[13:19:28.101] ERROR (32245): Error during template streaming
    request_id: "revle95fo7wx871avpuuxdfv"
    err: {
      "type": "Error",
      "message": "Échec de la génération du template à partir du document: Document content must be a URL (starting with 'https') or base64 encoded document (starting with 'data:<mime-type>;base64,<document-base64>')Received: 37,80,68,70,45,49,46,52,10,49,32,48,32,111,98,106,...114,101,102,10,56,53,48,53,55,10,37,37,69,79,70,10",
    }
```

Could you clarify:

Should I use a different field name (e.g., url instead of data) when passing a signed URL?
Or is the file content type only meant for base64-encoded content, requiring me to use a different approach for URL-based documents?

Thanks for your help!

### PR DESCRIPTION
**Pull request by graysonhicks**
## Description

Fixed `ModelRouterLanguageModel` to properly propagate `supportedUrls` from the underlying model provider. Previously, `supportedUrls` was hardcoded as an empty object `{}`, which caused Mastra to unnecessarily download files from URLs even when the model provider supports them natively.

### Root Cause

When using model strings like `"mistral/mistral-large-latest"` or `"openai/gpt-4o"`, Mastra creates a `ModelRouterLanguageModel` wrapper. This wrapper had:

```typescript
readonly supportedUrls = {} as Record<string, RegExp[]>;
```

This empty object meant that `isUrlSupported()` always returned `false`, causing all file URLs to be downloaded and converted to bytes/base64 before being sent to the model—even when the model supports URLs natively.

### The Fix

1. **`router.ts`**: Changed `supportedUrls` from a static empty object to a lazy `PromiseLike` that resolves the underlying model's `supportedUrls`. Added `_resolveSupportedUrls()` and `_fetchSupportedUrls()` methods to lazily fetch and cache the result.

2. **`llm-execution-step.ts`**: Updated to properly await `supportedUrls` when it's a Promise before passing to `downloadAssetsFromMessages()`.

### Impact

This fix benefits **all providers** that define native URL support:

| Provider | Supported URLs | Before Fix | After Fix |
|----------|---------------|------------|-----------|
| **OpenAI** (chat) | `image/*` via HTTP/HTTPS | ❌ Downloaded | ✅ Passed directly |
| **OpenAI** (responses) | `image/*` + `application/pdf` | ❌ Downloaded | ✅ Passed directly |
| **Anthropic** | `image/*` via HTTP/HTTPS | ❌ Downloaded | ✅ Passed directly |
| **Mistral** | `application/pdf` via HTTPS | ❌ Downloaded | ✅ Passed directly |
| **Google** | Files from Google endpoints | ❌ Downloaded | ✅ Passed directly |

### Benefits

- **Reduced latency**: No unnecessary file downloads
- **Reduced bandwidth**: Files aren't downloaded to Mastra server then re-uploaded
- **Fixes Mistral PDF/OCR**: Mistral expects PDF URLs, not raw bytes
- **Fixes signed URL expiration issues**: URLs are passed directly instead of being fetched (which could fail if the signed URL expires during processing)

## Related Issue(s)

Fixes #12152

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [x] Performance improvement
- [x] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works

## Summary by CodeRabbit

* **New Features**
  * URL support is now resolved lazily from underlying model providers, with provider-specific URL handling and caching to improve reliability and performance.
  * Providers that natively support URLs now pass them through directly.

* **Documentation**
  * Added guidance on URL accessibility requirements, fallback behavior when URL resolution fails, and edge cases for URL-based content downloads.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/core/src/llm/model/router-supported-urls.test.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import { ModelRouterLanguageModel } from './router';
+import * as gatewaysModule from './gateways/index';
+
```
[the remark this answers is not in the document]
Comment (graysonhicks) [reply packages/core/src/llm/model/router-supported-urls.test.ts]: fixed

**Review by wardpeet — APPROVED** [review]
